beacon2-import.py

```diff
@@ -0,0 +1,641 @@
+            filename (str): output filename including complete path
+
+        Returns:
+            Nothing
+
+    """
+    try:
+        gi.datasets.download_dataset(dataset.id, filename, use_default_filename=False)
+    except Exception as e:
+        # TODO catch exceptions
+        logging.critical(f"something went wrong while downloading file - {e}")
+
+
+def import_to_mongodb(BFF_files: dict):
+    """
+    Import a dataset to beacon
+
+        Parameters:
+            BFF_files (str): full path to the Beacon frindly files
+
+        Returns:
+            Nothing
+
+        Note:
+            This function uses MongoDB from the beacon2-ri-api package found at https://github.com/EGA-archive/beacon2-ri-api/tree/master/deploy
+
+            The connection settings are configured by ENVIRONMENT_VARIABLE or  "default value" if not set (not sure about the values need to check)
+
+                host: DATABASE_URL / "???",
+                port: DATABASE_PORT / "???",
+                user: DATABASE_USER / ???",
+                password: DATABASE_PASSWORD / beacon",
+                database: DATABASE_NAME / mongodb",
+
+    """
+
+    Entities = ['analyses_file', 'biosamples_file', 'cohorts_file', 'datasets_file', 'genomicVariations_file',
+                'individuals_file', 'runs_file']
+    global db
+    # Import  BFF files
+    for key in BFF_files.keys():
+        with open(BFF_files[key]) as f:
+            data = json.load(f)
+            collection_name = key
+            db.client.beacon.beacon[collection_name].insert_many(data)
+
+
+
+async def persist_variant_origins(dataset_id: str, dataset: str, record):
+    """
+    Maps dataset_id to variant index in a separate file (which is hard-coded)
+
+        Note:
+            We do not want any information in the beacon database that may be used to reconstruct an actual file
+            uploaded to galaxy. Additionally, we do not want to change the dataset import functions provided by
+            beacon python.
+            Therefore, a separate file is maintained linking variant indices and dataset IDs. Since we do not interfer
+            with the actual variant import these indices have to be queried individually from beacons database.
+
+        Parameters:
+            dataset_id (str): Dataset id as returned by the galaxy api
+            dataset (VCF): The actual dataset
+            record (Any): Output file in which to persist the records
+
+        Returns:
+            Nothing.
+    """
+    global db
+    with open(dataset) as j_f:
+        data = json.load(j_f)
+        for variant in data:
+            ALT = variant['alternateBases']
+            start = variant['position']['start'][0]
+            REF = variant['referenceBases']
+            res=db.client.beacon.genomicVariations.find({"alternateBases":ALT},{"start":start},{"referenceBases":REF})
+            record.write(f'{res["_id"]}{dataset_id}\n')
+
+async def update_variant_counts():
+    """
+    asd

```

Comment:
ping

Reply:
ping

Reply:
For this part we were confused, 

Are we looking into the dataset counts from the input files? we changed the function (update_dataset_counts) to pass until we know/ask about it.

Reply:
For this, we are unsure of where the data was taken to create the list. Are they from inside the datasets or from the galaxy's info?

Reply:
Hi @bgruening 

This is the last part we are working on, and we know the API that creates this [beacon_dataset_counts_table](https://github.com/CSCfi/beacon-python/blob/e5c1b357244b9e8db9baff2e9e0f1061d8f076e3/beacon_api/utils/db_load.py) and that was used in the [original code](https://github.com/ELIXIR-UK-DaSH/galaxy-beacon-import/blob/c4be056330ee2b1271e0f1266133a6e63cfac552/beacon-import.py#L66)

I don't know from where they got the data (datasetid, callcount, variantcount) or if we need this part in beacon2 as we already have the data listed in JSON format